src/app/components/core/system-alert/system-alert.component.html

```diff
@@ -1,37 +1,37 @@
 <div id="dropdown" class="dropdown is-right" id="system-alert-dd">
   <div class="dropdown-trigger">
-    <a class="button is-small system-alert" aria-haspopup="true" aria-controls="dropdown-menu" (click)="toggleDropdown()">
+    <a class="button is-small system-alert" [title]="alertsCount === 0 ? 'No Alerts' : 'Alerts'" aria-haspopup="true" aria-controls="dropdown-menu" (click)="toggleDropdown()">
       <span class="icon icon-badge-container">
         <i class="bi bi-bell bi-lg" aria-hidden="true"></i>
         <div class="icon-badge">{{alertsCount}}</div>
       </span>
     </a>
   </div>
   <div class="dropdown-menu" id="dropdown-menu" role="menu">
-    <div class="dropdown-content">
+    <div class="dropdown-content" id="alerts">
       <ng-container *ngFor="let alert of systemAlerts; let i=index;">
-        <div class="columns dropdown-item pb-0">
-          <a class="column is-vertical-center is-1">
+        <div class="columns dropdown-item" [ngClass]="{'pb-0 mb-0': !rolesService.hasEditPermissions()}">
+          <span class="column is-vertical-center is-1">
             <i [ngClass]="applyClass(alert?.urgency)" class="bi bi bi-exclamation-triangle"></i>          
-          </a>
+          </span>
           <div class="column">
             <div class="message-text">
               {{alert.message}}
             </div>
           </div>
+          
         </div>
-        <div class="columns mb-0 mr-1">
-          <!-- TODO: Show timestamp -->
-          <!-- <div class="column is-1">
+        <div class="columns p-0 mb-0 mr-1">
+          <div class="column is-1">
+          </div>
+          <div class="column pt-1 pb-0 pl-5" [ngClass]="{'align-timestamp': !rolesService.hasEditPermissions()}">
+            <span class="help has-text-grey">{{getAlertTime(alert?.timestamp)}}</span>
           </div>
-          <div class="column is-pulled-right">
-            <span class="help">1m ago</span>
-          </div> -->
           <ng-container *ngIf="rolesService.hasEditPermissions()">
             <div class="column pt-0 pb-0">
               <div class="field is-grouped is-pulled-right">          
-                <p class="control">
-                  <button class="button is-tiny is-outlined is-link" (click)="performAction(alert)">{{getButtonText(alert?.message)}}</button>
+                <p *ngIf="['Show Logs', 'Upgrade'].includes(alert?.buttonText)" class="control">
```

Comment:
better to keep this static array in the ts file

Reply:
Agree, an object/map!
Let's do in next patch and get this in as is.